### DISCUSSION TITLE
postUpgradeTasks for specific dependencies

### DISCUSSION THREAD

**Discussion by 56KBs** [renovatebot/renovate#8735] [Request Help] [answered]
Is it possible to run a `postUpgradeTask` when specific dependencies are updated?

Use case:
Go 1.16 released, renovate raised MRs to update our Dockerfiles to use this release. Our CI system failed due to the `go.sum` file needing to be updated (Bugfixes in 1.16's release).

We'd like to be able to run `go mod tidy` when this Docker dependency is updated in future to avoid manually having to run `go mod tidy` for this kind of MRs

**Comment by rarkins** [comment]
Post upgrade tasks can be configured with package rules, meaning you can configure them per-package. If you are grouping together dependencies and want to run commands per-update in a branch then you'd need to wait for https://github.com/renovatebot/renovate/pull/8725 to land

**Comment by viceice** [comment, marked as answer]
@56KBs [postUpdateOptions](https://docs.renovatebot.com/configuration-options/#postupdateoptions) should automatically run `gomodTidy` if you configure it. It's a renovate buildin feature.
